src/dimensions/fixedmass.jl

```diff
@@ -0,0 +1,50 @@
+using SpecialFunctions
+
+"""
+    fixed_mass_algorithm(data, max_j; metric = Euclidean(), M = length(data)) → rs, ϵs
+A fixed mass algorithm for the calculation of the fractal dimension according
+to [^Grassberger 1988] with `max_j` the maximum number of neighbours that
+should be considered for the calculation, `M` defines the number of points
+considered for the calculation, default is the whole data set.
+
+Implements
+```math
+D_q \\overline{\\log r^{(j)}} \\sim Ψ(j) - \\log N
+```
+where `` Ψ(j) = \\frac{\\text{d} \\log Γ(j)}{\\text{d} j}
+``, `rs` = ``\\overline{\\log r^{(j)}}`` and `ϵs` = ``Ψ(j) - \\log N``.
+
+``D_q`` can be computed by using `linear_region(rs, ϵs)`.
+
+[^Grassberger 1988]: Peter Grassberger (1988) [Finite sample Corrections to Entropy and Dimension Estimates, Physics Letters A 128(6-7)](https://doi.org/10.1016/0375-9601(88)90193-4)
+"""
+function fixed_mass_algorithm(data, max_j; metric = Euclidean(), M = length(data))
+    # Define the length of the data set.
+    N = length(data)
+    # Transform the data into a tree.
+    tree = KDTree(data, metric)
+    # Calculate all nearest neighbours up to max_j.
+    _, distances =
+        knn(
+            tree,
+            # If M is given, only M random points of the set are considered.
+            M == N ?
+                [point for point in data] :
+                [data[index] for index in randperm(N)[1:M]],
```

Comment:
```suggestion
                data[view(randperm(N), 1:M)],
```

this is a critical performance improvement you should be aware about (view)